src/Validation/ValidationRule.php

```diff
@@ -40,7 +40,7 @@ class ValidationRule
      *
      * @var callable|string|null
      */
-    protected $_on = null;
+    protected $_on;
```

Comment:
This doesn't make sense to me. Removing the null initialization will cause an `not initialized error` if the property is accessed without explicitly setting a value.

Reply:
Both variables are same: https://stackoverflow.com/questions/14097258/is-a-property-default-value-of-null-the-same-as-no-default-value

Reply:
Not if you set a type for it. Which here is the case.

Reply:
Isn't this typed property? public int $id;

Reply:
Ah yes, this is an untyped property since `callable` can't be used.